### PR TITLE
ROC-2508: Switch to new draft store

### PR DESCRIPTION
**Pull request by dharmendrak**

### CODE REVIEW

File: src/main/app/drafts/models/draftLegalClaim.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ export default class DraftLegalClaim implements Serializable<DraftLegalClaim> {
 
   deserialize (input: any): DraftLegalClaim {
     if (input) {
-      this.externalId = input.externalId
+      this.externalId = input['externalId']
```
Comment (timja-kainos) [inline src/main/app/drafts/models/draftLegalClaim.ts]: this is wrong, it should be on a class

File: src/main/common/draft/draftMiddleware.ts [hunk 2]
```diff
@@ -1,41 +1,60 @@
...
+export class DraftMiddleware {
 
-  delete (res: express.Response, next: express.NextFunction): Promise<void> {
-    return this.client
-      .delete(res.locals.user.id)
+  static requestHandler<T> (draftType: string, deserializeFn: (value: any) => T = (value) => value): express.RequestHandler {
+    return async function (req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> {
+      if (res.locals.isLoggedIn) {
+        try {
+          new DraftService()['find'](draftType, '100', res.locals.user.bearerToken, deserializeFn)
```
Comment (timja-kainos) [inline src/main/common/draft/draftMiddleware.ts]: this is also wrong 
```
new DraftService().find(...)
```

File: src/main/features/claim/routes/claimant-change.ts [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 
 export default express.Router()
 
   .get(Paths.claimantChangePage.uri,
     ErrorHandling.apply(async (req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> => {
 
-      res.locals.user.viewDraft.claimantChangeIndex = Claimants.getChangeIndex(req, res)
-      await ViewDraftMiddleware.save(res, next)
+      res.locals.user.viewDraft.document.claimantChangeIndex = Claimants.getChangeIndex(req, res)
+      await new DraftService()['save'](res.locals.user.viewDraft, res.locals.user.bearerToken)
```
Comment (timja-kainos) [inline src/main/features/claim/routes/claimant-change.ts]: same with this

File: src/main/app/drafts/models/draftLegalClaim.ts (hunk 1 above)
Comment (sabahirfan) [reply src/main/app/drafts/models/draftLegalClaim.ts]: Done.

Comment (dharmendrak) [reply src/main/app/drafts/models/draftLegalClaim.ts]: it's not wrong, we have removed DraftDocument class

Comment (dharmendrak) [reply src/main/app/drafts/models/draftLegalClaim.ts]: And external Id is draft JavaScript object

File: src/main/common/draft/draftMiddleware.ts (hunk 2 above)
Comment (dharmendrak) [reply src/main/common/draft/draftMiddleware.ts]: We are calling JavaScript functions here

File: src/main/features/claim/routes/claimant-change.ts (hunk 3 above)
Comment (dharmendrak) [reply src/main/features/claim/routes/claimant-change.ts]: Same here calling JavaScript function

Comment (timja-kainos) [reply src/main/features/claim/routes/claimant-change.ts]: still wrong and shouldn't be doing it

File: src/main/common/draft/draftMiddleware.ts (hunk 2 above)
Comment (timja-kainos) [reply src/main/common/draft/draftMiddleware.ts]: still wrong and shouldn't be doing it

File: src/main/app/drafts/models/draftLegalClaim.ts (hunk 1 above)
Comment (timja-kainos) [reply src/main/app/drafts/models/draftLegalClaim.ts]: still wrong and shouldn't be doing it

File: package.json [hunk 4]
```diff
@@ -14,7 +14,7 @@
     "setup": "gulp sass copy-files",
     "lint": "tslint --project tsconfig.json --type-check && sass-lint -v -q",
     "test": "NODE_ENV=mocha LOG_LEVEL=OFF mocha --compilers ts:ts-node/register -r tsconfig-paths/register $(find src/test \\( -name '*.ts' -or -name '*.js' \\) ! -path '*a11y*' ! -path '*/routes/*') --reporter mochawesome --reporter-options reportFilename=unit,inlineAssets=true,reportTitle=legal-rep-frontend-unit",
-    "test:routes": "NODE_ENV=mocha LOG_LEVEL=OFF mocha --compilers ts:ts-node/register -r tsconfig-paths/register $(find src/test \\( -name '*.ts' -or -name '*.js' \\) -path '*/routes/*') --reporter mochawesome --reporter-options reportFilename=routes,inlineAssets=true,reportTitle=legal-rep-frontend-routes",
+    "test:routes": "NODE_ENV=mocha LOG_LEVEL=OFF mocha --compilers ts:ts-node/register -r tsconfig-paths/register $(find src/test \\( -name '*.ts' -or -name '*.js' \\) -path '*/routes/defendant*') --reporter mochawesome --reporter-options reportFilename=routes,inlineAssets=true,reportTitle=legal-rep-frontend-routes",
```
Comment (timja-kainos) [inline package.json]: why are you excluding defendant route tests?

Comment (dharmendrak) [reply package.json]: good one, was testing, got left. thanks

**Pull request merged**